packages/contracts-bedrock/test/L1/OPContractsManager.t.sol

```diff
@@ -2083,3 +2101,134 @@ contract OPContractsManager_SetRC_Test is OPContractsManager_Upgrade_Harness {
         opcm.setRC(true);
     }
 }
+
+/// @title OPContractsManager_L1ContractsRelease_Test
+/// @notice Tests the `l1ContractsRelease` function of the `OPContractsManager` contract.
+contract OPContractsManager_L1ContractsRelease_Test is OPContractsManager_TestInit {
+    /// @notice Tests that l1ContractsRelease returns the correct value when isRC is true.
+    function test_l1ContractsRelease_rcTrue_succeeds() public view {
+        // isRC is true by default
+        string memory release = opcm.l1ContractsRelease();
+        bytes memory releaseBytes = bytes(release);
+        assertEq(Bytes.slice(releaseBytes, releaseBytes.length - 3, 3), "-rc", "release should end with '-rc'");
+    }
+}
+
+/// @title OPContractsManager_Validate_Test
+/// @notice Tests the `validate` function of the `OPContractsManager` contract.
+contract OPContractsManager_Validate_Test is OPContractsManager_TestInit {
+    /// @notice Tests that validate function returns validation results.
+    function test_validate_succeeds() public view {
+        // Create a validation input using deployed contracts
+        IOPContractsManagerStandardValidator.ValidationInput memory input;
+        input.l2ChainID = 100; // Using the l2ChainId from chainDeployOutput1
+        input.sysCfg = chainDeployOutput1.systemConfigProxy;
+        input.proxyAdmin = chainDeployOutput1.opChainProxyAdmin;
+        input.absolutePrestate = Claim.wrap(bytes32(hex"ABCDABCD")).raw();
+
+        string memory result = opcm.validate(input, true);
+        // The result should be a string (even if empty or with errors)
+        assertTrue(bytes(result).length >= 0, "validate should return a result");
+    }
+
+    /// @notice Tests validate with allowFailure set to false reverts when validation fails.
+    function test_validate_allowFailureFalse_reverts() public {
+        IOPContractsManagerStandardValidator.ValidationInput memory input;
+        input.l2ChainID = 101; // Using the l2ChainId from chainDeployOutput2
+        input.sysCfg = chainDeployOutput2.systemConfigProxy;
+        input.proxyAdmin = chainDeployOutput2.opChainProxyAdmin;
+        input.absolutePrestate = Claim.wrap(bytes32(hex"DEADBEEF")).raw();
+
+        // When allowFailure is false and validation fails, it should revert
+        // The error message contains validation failure codes
+        vm.expectRevert(
```

Comment:
Meh this test will probably get annoying, maybe we can just vm.expectRevert without any error string (you will need to have semgrep ignore it).